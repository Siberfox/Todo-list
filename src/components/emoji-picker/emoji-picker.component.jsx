import React from "react";

import "./emoji-picker.styles.scss";

export const emoji = [
  { id: 1, label: "Document", sign: "📝" },
  { id: 2, label: "Bed", sign: "🛌" },
  { id: 3, label: "Time", sign: "⌚" },
  { id: 4, label: "Knife", sign: "🔪" },
  { id: 5, label: "Bomb", sign: "💣" },
  { id: 6, label: "Holiday", sign: "🎉" },
  { id: 7, label: "Gift", sign: "🎁" },
  { id: 8, label: "Phone", sign: "📞" },
  { id: 9, label: "Money", sign: "💰" },
  { id: 10, label: "Shop cart", sign: "🛒" },
];

export const EmojiPicker = ({ addEmoji }) => {
  return (
    <div className="emoji-picker">
      {emoji.map((item) => (
        <span
          className="emoji-item"
          onClick={addEmoji}
          role="img"
          aria-label={item.label}
          key={item.id}
        >
          {item.sign}
        </span>
      ))}
    </div>
  );
};
