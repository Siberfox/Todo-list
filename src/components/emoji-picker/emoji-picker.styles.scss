.emoji-picker {
  position: absolute;
  top: 35px;
  height: 30px;
}
.emoji-item {
  font-size: 22px;
  margin-right: 6px;
}
