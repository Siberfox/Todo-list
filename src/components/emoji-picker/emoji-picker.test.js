import React from "react";
import { shallow } from "enzyme";

import { EmojiPicker, emoji } from "./emoji-picker.component";

describe("Emoji picker", () => {
  const mockAddEmoji = jest.fn();
  const wrapper = shallow(<EmojiPicker addEmoji={mockAddEmoji} />);

  it("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("emoji list length should be equal emoji array length", () => {
    expect(wrapper.find(".emoji-picker").children().length).toEqual(
      emoji.length
    );
  });

  it("should call addEmoji on items click", () => {
    wrapper.find(".emoji-item").at(1).simulate("click");
    expect(mockAddEmoji).toHaveBeenCalled();
  });
});
