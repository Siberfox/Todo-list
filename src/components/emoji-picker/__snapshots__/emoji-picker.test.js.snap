// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji picker should render component 1`] = `
<div
  className="emoji-picker"
>
  <span
    aria-label="Document"
    className="emoji-item"
    key="1"
    onClick={[MockFunction]}
    role="img"
  >
    📝
  </span>
  <span
    aria-label="Bed"
    className="emoji-item"
    key="2"
    onClick={[MockFunction]}
    role="img"
  >
    🛌
  </span>
  <span
    aria-label="Time"
    className="emoji-item"
    key="3"
    onClick={[MockFunction]}
    role="img"
  >
    ⌚
  </span>
  <span
    aria-label="Knife"
    className="emoji-item"
    key="4"
    onClick={[MockFunction]}
    role="img"
  >
    🔪
  </span>
  <span
    aria-label="Bomb"
    className="emoji-item"
    key="5"
    onClick={[MockFunction]}
    role="img"
  >
    💣
  </span>
  <span
    aria-label="Holiday"
    className="emoji-item"
    key="6"
    onClick={[MockFunction]}
    role="img"
  >
    🎉
  </span>
  <span
    aria-label="Gift"
    className="emoji-item"
    key="7"
    onClick={[MockFunction]}
    role="img"
  >
    🎁
  </span>
  <span
    aria-label="Phone"
    className="emoji-item"
    key="8"
    onClick={[MockFunction]}
    role="img"
  >
    📞
  </span>
  <span
    aria-label="Money"
    className="emoji-item"
    key="9"
    onClick={[MockFunction]}
    role="img"
  >
    💰
  </span>
  <span
    aria-label="Shop cart"
    className="emoji-item"
    key="10"
    onClick={[MockFunction]}
    role="img"
  >
    🛒
  </span>
</div>
`;
